q2_types/per_sample_sequences/_format.py

```diff
@@ -105,25 +107,37 @@ class FastqGzFormat(model.BinaryFileFormat):
     A gzipped fastq file.
 
     """
-    def sniff(self):
+    def validate(self):
         with self.open() as fh:
-            if fh.read(2)[:2] != b'\x1f\x8b':
-                return False
-
-        filepath = str(self)
-        sniffer = skbio.io.io_registry.get_sniffer('fastq')
-        if sniffer(str(self))[0]:
-            try:
-                generator = skbio.io.read(filepath, constructor=skbio.DNA,
-                                          phred_offset=33, format='fastq',
-                                          verify=False)
-                for seq, _ in zip(generator, range(15)):
-                    pass
-                return True
-            # ValueError raised by skbio if there are invalid DNA chars.
-            except ValueError:
-                pass
-        return False
+            if fh.peek(2)[:2] != b'\x1f\x8b':
+                raise ValidationError('File is uncompressed')
+
+        with gzip.open(str(self), mode='rt', encoding='ascii') as fh:
+            for i, record in enumerate(itertools.zip_longest(*[fh] * 4)):
+                header, seq, sep, qual = record
+
+                if not header.startswith('@'):
+                    raise ValidationError('Header on line %d is not FASTQ, '
+                                          'records may be misaligned' %
+                                          (i * 4 + 1))
+
+                if seq is None:
+                    raise ValidationError('Missing sequence for record '
+                                          'beginning on line %d'
+                                          % (i * 4 + 1))
+                elif not seq.isupper():
+                    raise ValidationError('Lowercase case sequence on line %d'
+                                          % (i * 4 + 2))
+
+                if sep is None:
```

Comment:
Do you think it's worth checking that the separator line starts with `+`?

Reply:
The `@` should trip up in the next record, so we thought we could save an operation in this loop.

Reply:
I mean for validation purposes (it's part of the fastq spec)

Reply:
Following up from slack. It's possible for *something else* to be here, which would trip things up. The FASTQ header is allowed to be repeated, so we should use `sep.startswith('+')` as an `else` block.